format/private/format.sh

```diff
@@ -142,6 +142,7 @@ function ls-files {
       'TypeScript') patterns=('*.ts' '*.cts' '*.mts') ;;
       'Vue') patterns=('*.vue') ;;
       'YAML') patterns=('*.yml' '*.yaml' '.clang-format' '.clang-tidy' '.gemrc') ;;
+      'XML') patterns=('*.xml') ;;
```

Comment:
hm, so far I've been mirroring what GitHub considers to be the filename patterns for a given language. Not surprising that XML is a giant list though https://github.com/github-linguist/linguist/blob/559a6426942abcae16b6d6b328147476432bf6cb/lib/linguist/languages.yml#L7767-L7882

Somehow they also permit duplicates, like `".tsx"` is in this list.

I recognize a couple of these as things we should include here (like ".xsd") but it will take a little extra thought, and it's sad to have to make such a decision here instead of blindly following a "trusted" external source

Reply:
That looks like a great approach to me. I had been thinking of encoding the extensions in starlark but JSON is a lot simpler. I created #484 with your commit and I'll do a bit more on top of it. Thanks!